akka-projection-core/src/main/scala/akka/projection/Projection.scala

```diff
@@ -67,10 +68,19 @@ trait Projection[Envelope] {
 @InternalApi
 private[projection] object RunningProjection {
 
+  /**
+   * When stopping an projection the retry mechanism is aborted via this exception.
+   */
+  case object AbortProjectionException extends RuntimeException("Projection aborted.") with NoStackTrace
```

Comment:
Interesting.  Neat trick for signalling failure.  I could have done this in the offset verification instead of introducing new types.